Backend/app/api/v1/auth_api.py

```diff
@@ -0,0 +1,70 @@
+import os
+from dotenv import load_dotenv
+from fastapi import APIRouter, Depends, HTTPException, Response
+from fastapi.security import OAuth2PasswordBearer, OAuth2PasswordRequestForm
+from app.repository.user_repository import UserRepository
+from app.dependency.repository import get_user_repository
+from datetime import datetime, timedelta
+from jose import JWTError, jwt
+
+router = APIRouter()
+oauth2_scheme = OAuth2PasswordBearer(tokenUrl="token")
+
+load_dotenv()
+SECRET_KEY = os.getenv("SECRET_KEY")
+ALGORITHM = "HS256"
+
+
+def create_access_token(data: dict, expires_delta: timedelta = None):
+    to_encode = data.copy()
+    if expires_delta:
+        expire = datetime.utcnow() + expires_delta
+    else:
+        expire = datetime.utcnow() + timedelta(minutes=15)
+    to_encode.update({"exp": expire})
+    encoded_jwt = jwt.encode(to_encode, SECRET_KEY, algorithm=ALGORITHM)
+    return encoded_jwt
+
+
+@router.post("/token")
+async def login_for_access_token(
+    response: Response,
+    form_data: OAuth2PasswordRequestForm = Depends(),
+    user_repo: UserRepository = Depends(get_user_repository),
+):
+    # check userdata
+    is_authenticated = user_repo.authenticate_user(
+        email=form_data.username, password=form_data.password
+    )
+    if not is_authenticated:
+        raise HTTPException(
+            status_code=400,
```

Comment:
Use 401 Not authenticated instead of 400 and there is also a package providing an enum for all common status codes. Have a look at the other api.

Reply:
see commit, nice suggestion